challenges/sre/challenge-security.md

```diff
@@ -0,0 +1,202 @@
+
+### Service Levels & Deployment
+
+After the server code is approved, submit a final PR including service levels, deployment, and any remaining requirements.
+
+Your chosen service levels should focus on meaningful indicators, be set at realistic thresholds, and align with the system's functionality and user experience.  The presentation of these SLOs is up to the candidate's discretion.  Consider a simple solution and include comments on how a more robust solution could be implemented (see [trade-offs](#trade-offs)).
+
+When selecting [security invariants](https://d1.awsstatic.com/events/aws-reinforce-2022/NIS234_Learn-to-create-end-to-end-security-invariants-using-AWS-services.pdf), consider which properties you expect your system to always maintain.  The chosen invariants should cover attack vectors against the most critical parts of your system design. How these properties are evaluated and reported is up to the candidate.
+
+This part of the implementation will be evaluated ensuring the helm chart includes production-level packaging, service to service communication is secured implementing zero-trust, all artifacts are easy to build and deploy, and that the selected security invariants of the system can be evaluated.
+
+### Tooling
+For evaluation purposes, your solution should be written in Go and is deployable to a local Kubernetes cluster. The choice of which local Kubernetes cluster is up to you, but please ensure compatibility with both macOS and Linux. We suggest [KIND](https://kind.sigs.k8s.io/).
+
+You may use additional external dependencies, but ensure that detecting or installing these are straightforward for the reviewer. At a minimum, your solution should include the following dependencies:
+
+* Go
+* make
+* Docker
+
+## Level 4
+
+### Server
+* HTTPS API to create, get, or list users with Redis datastore
+* HTTPS API to add a new payment with Redis datastore
+* HTTPS API to get payments by user with Redis datastore
+* HTTPS Health check ensuring availability
+* Create a secure client API with mTLS
+* Include one or two tests that cover happy and unhappy path scenarios
+
+### Service Levels
+* Define three Service Level Objectives (SLOs) for the API service
+* Create a service to periodically evaluate the defined SLOs and report on their status
+* SLO service communication is secure
+
+### Deployment
+* Create a Helm chart for the API server
+* Include the ability to build and deploy all artifacts and certificates to a Kubernetes cluster using make
+* Ensure artifacts and services are production ready
+* Evaluate the state of three security invariants.
```

Comment:
How do you think they will implement these security invariants given the details here?

Reply:
Many possible approaches with different pros/cons
- Make target that evaluates current state upon execution
- Cron job to evaluate state periodically (could just log the output)
- Expose the state via API, could be an extension of SLO service or separate

As the instructions (hopefully) hint, the main focus is on understanding invariants and identifying important attack vectors to their service.